### PR TITLE
Add country ip lookup route and component

### PR DESCRIPTION
**Pull request by AndreasPB**
# Description

* Adds country lookup through ip

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### CODE REVIEW

File: backend/app/routers/country.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+router = APIRouter(
+    prefix="/country",
+    tags=["country"],
+    responses={404: {"description": "Country not found"}},
+)
+
+
+@router.get("/")
+async def read_all_genres(request: Request):
```
Comment (Pankai222) [inline backend/app/routers/country.py:17]: Am I missing something, or should this endpoint be named something else?

Comment (AndreasPB) [reply backend/app/routers/country.py:17]: Oh haha, you're not, I just forgot to rename it :D Should be `lookup_country()` or something like that

File: frontend/src/components/countryLocator.svelte [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  import { currentCountry } from "../stores/country"
+  import { PYTHON_API } from "../variables"
+
+  const lookupCountry = async () => {
+    await fetch(`${PYTHON_API}/country`)
+      .then(async (response: Response) => {
+        const data = await response.json()
+
+        if (!response.ok) {
+          alert("Failed to lookup country")
```
Comment (Pankai222) [inline frontend/src/components/countryLocator.svelte:13]: Alerts are so 2004. Should maybe be done another way as to not confuse the user.

Comment (AndreasPB) [reply frontend/src/components/countryLocator.svelte:13]: Yea that was mostly for debugging - I just need people to test if the thing works. I have a hard time figuring out the corner cases

Comment (Pankai222) [reply frontend/src/components/countryLocator.svelte:13]: Maybe a modal with some text along the lines of: "We could not automatically detect which country you're in. Use the blahblahblah to choose your country."

Comment (Pankai222) [reply frontend/src/components/countryLocator.svelte:13]: Alright :^)

Comment (AndreasPB) [reply frontend/src/components/countryLocator.svelte:13]: Yea, a modal could be nice

File: frontend/src/components/countryLocator.svelte [hunk 3]
```diff
@@ -0,0 +1,69 @@
+<script lang="ts">
+  import { onMount } from "svelte"
+  import { patron } from "../stores/patron"
+  import { currentCountry } from "../stores/country"
+  import { PYTHON_API } from "../variables"
+
+  let hasError: boolean = false
```
Comment (Primdahl26) [inline frontend/src/components/countryLocator.svelte:7]: isnt telling its a `boolean` here redundant?

**Review by Primdahl26 — COMMENTED** [review]
I feel like we maybe should have a timer that it goes away by itself after?

Also i feel like we should consider this being sticky and be a little bit transparent?

LMK what your thoughts are

File: frontend/src/stores/patron.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import { persist, localStorage } from "@macfja/svelte-persistent-store"
+import { writable } from "svelte/store"
+
+export const patron = persist(writable(false), localStorage(), "patron")
```
Comment (Primdahl26) [inline frontend/src/stores/patron.js:4]: I hate this name, can it be something else? `patron` is someone that supports us financially. Would it not make sense with something like `country_confirmed`, `has_country`, or something else in that direction.

Comment (AndreasPB) [reply frontend/src/stores/patron.js]: I changed it to `confirmedCountry` and moved it the country.js store file

I will need you to test something that is super confusing tho :D The page works after the change, but it seems like the deleted `patron` localstorage is still being used instead of the new name... I can't get it to change

File: frontend/src/components/countryLocator.svelte (hunk 3 above)
Comment (AndreasPB) [reply frontend/src/components/countryLocator.svelte]: I guess

File: frontend/src/stores/patron.js (hunk 4 above)
Comment (Primdahl26) [reply frontend/src/stores/patron.js]: Very nice

**Review by Primdahl26 — APPROVED** [review]
